funcbench/README.md

```diff
@@ -28,8 +28,53 @@ The benchmark can be triggered by creating a comment in a PR which specifies a b
 /funcbench <branch/commit> <benchmark function regex>
 ```
 
+#### Setup
+
 The comment is handled by [comment-monitor](https://github.com/prometheus/test-infra/tree/master/tools/commentMonitor) and then the parsed arguments are handed over to funcbench(if using Github Actions) or to [prombench](https://github.com/prometheus/test-infra/tree/master/prombench) if using funcbench with GKE.
 
+- Create GitHub actions workflow file (see below) that is executed when an `repository_dispatch` event is on.
+- Read BRANCH / PR_NUMBER / REGEX from event payload into environment variables.
+
+#### Example GitHub action workflow file
+
+```yaml
+on: repository_dispatch
+name: Funcbench Workflow
+jobs:
+  run_funcbench:
+    name: Running funcbench
+    if: github.event.action == 'funcbench_start'
+    runs-on: ubuntu-latest
+    env:
+      AUTH_FILE: ${{ secrets.PROMBENCH_GKE_AUTH }}
+      CLUSTER_NAME: << cluster name >>
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      PROJECT_ID: << project id >>
+      PR_NUMBER: ${{ github.event.client_payload.PR_NUMBER }}
+      TEST_INFRA_REPO: https://github.com/prometheus/test-infra.git # put your repo here if test your own test-infra
+      ZONE: << zone >>
+    steps:
+    - name: Prepare nodepool
+      uses: docker://prominfra/prombench:latest
+      with:
+        args: make funcbench_nodepool_create
+    - name: Run funcbench
+      uses: docker://prominfra/prombench:latest
+      env:
+        BRANCH: ${{ github.event.client_payload.BRANCH }}
+        GITHUB_ORG: prominfra
+        GITHUB_REPO: prometheus
+        GITHUB_TOKEN: ${{ secrets.PERSONAL_TOKEN }} # if benchmark lasts more than 60 minutes, have to create your own token, and put it into PERSONAL_TOKEN
+        REGEX: ${{ github.event.client_payload.REGEX }}
```

Comment:
hm, why not provide the regex directly here? It will be easier to change and understand.

Reply:
Since it's an example triggered by `repository_dispatch` event. ;)

Reply:
I still don't understand why do we need to specify it elsewhere and not directly here?

Reply:
`REGEX` is an input from user, we need comment-monitor to extract it out, then pass it to GH action via `repository_dispatch` event.
Do you suggest switching back to `issue_comment` event ?